rskj-core/src/main/java/org/ethereum/core/Block.java

```diff
@@ -690,16 +700,14 @@ public void setBitcoinMergedMiningCoinbaseTransaction(byte[] bitcoinMergedMining
     }
 
     public static Trie getTxTrie(List<Transaction> transactions){
-        Trie txsState = new TrieImpl();
-        int itran = 0;
-
-        if (transactions != null) {
-            for (int i = 0; i < transactions.size(); i++) {
-                Transaction transaction = transactions.get(i);
+        if (transactions == null) {
+            return new TrieImpl();
+        }
 
-                txsState.put(RLP.encodeInt(itran), transaction.getEncoded());
-                itran++;
-            }
+        Trie txsState = new TrieImpl();
+        for (int i = 0; i < transactions.size(); i++) {
+            Transaction transaction = transactions.get(i);
+            txsState = txsState.put(RLP.encodeInt(i), transaction.getEncoded());
```

Comment:
Bug like this are really hard to spot, and we should think about turning on some more Sonar rules to help in these cases.